massgen/frontend/displays/textual_terminal_display.py

```
@@ -2406,6 +2406,58 @@ def handle_result(result: ReviewResult) -> None:
+        result_future: asyncio.Future = loop.create_future()
+
+        def show_modal():
+            try:
+                modal = ToolApprovalModal(authz)
+
+                def handle_result(result) -> None:
+                    if not result_future.done():
+                        loop.call_soon_threadsafe(result_future.set_result, result)
+
+                self._app.push_screen(modal, handle_result)
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)